blockprotocol/blockprotocol
# [FE] Add search to docs pages

## What this Does

The blockprotocol website currently has its documentation spread across the [docs](https://blockprotocol.org/docs) and [spec](https://blockprotocol.org/spec) pages. While this data is easily accessible via the navbar, a new user may find it difficult to search for a specific part of the documentation. 

This PR [adds code to keep an Algolia Index updated](https://github.com/blockprotocol/blockprotocol/pull/125/commits/9f15dc0861647021692bd5640a73a3834490186f) with the latest data, and [adds a Search Component](https://github.com/blockprotocol/blockprotocol/pull/125/commits/c6e4b83afb64f88bb711a3e7566329710f66c2be) to allow this data to be searchable within the app. The Algolia data is kept updated based on any changes made to the `_pages` files, and deletes any removed pages as needed. 

This is similar to how the Algolia updating logic works in the [hash repo](https://github.com/hashintel/hash/tree/main/.github/workflows/algolia).

## Follow-ups

- Update styles to match mui way of styling things.
- Add `ALGOLIA_PROJECT` and `ALGOLIA_WRITE_KEY` to the GitHub repository
- Create mobile version of the search bar

## How to test it

1. Go to /docs or /spec
2. Play around with the search bar

![01UgEpKGWq](https://user-images.githubusercontent.com/22196279/150995321-77c927f8-569e-43c5-a3fc-7a80c4af285b.gif)




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201620599409792
